### PR TITLE
[TIP] Integrate Threat Intelligence plugin with Security Solution

### PR DESCRIPTION
**Pull request by PhilippeOberti**
### Summary

This PR integrates the Threat Intelligence plugin in the Security Solution plugin.

No link was added to the main flyout left navigation, only to the Security sidebar navigation:
- in the old navigation, link was added in the Explore section below Users
- in the new navigation, link was added as a panel in the Explore page

https://github.com/elastic/security-team/issues/4329

#### Notes: this PR will conflict with https://github.com/lgmys/kibana/pull/5, I can fix the conflict after the later is merged.

https://user-images.githubusercontent.com/17276605/177877738-9fd5056b-af79-4dda-862a-761a94666ee5.mov

### CODE REVIEW

File: x-pack/plugins/security_solution/public/app/deep_links/index.ts [hunk 1]
```diff
@@ -372,6 +374,18 @@ export const securitySolutionsDeepLinks: SecuritySolutionDeepLink[] = [
           },
         ],
       },
+      {
+        id: SecurityPageName.threatIntelligence,
+        title: THREAT_INTELLIGENCE_RESPONSE,
+        path: THREAT_INTELLIGENCE_PATH,
+        navLinkStatus: AppNavLinkStatus.hidden,
```
Comment (PhilippeOberti) [inline x-pack/plugins/security_solution/public/app/deep_links/index.ts:381]: this hides the link entry in the main Kibana navigation

File: x-pack/plugins/security_solution/public/threat_intelligence/links.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { SecurityPageName, SERVER_APP_ID, THREAT_INTELLIGENCE_PATH } from '../../common/constants';
+import { THREAT_INTELLIGENCE_RESPONSE } from '../app/translations';
+import { LinkItem } from '../common/links';
+import threatIntelligencePageImg from '../common/images/threat_intelligence.png';
+
+export const links: LinkItem = {
```
Comment (PhilippeOberti) [inline x-pack/plugins/security_solution/public/threat_intelligence/links.ts:14]: this is what's used to show the panel in the Explore page in the new navigation

File: x-pack/plugins/security_solution/public/threat_intelligence/links.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { SecurityPageName, SERVER_APP_ID, THREAT_INTELLIGENCE_PATH } from '../../common/constants';
+import { THREAT_INTELLIGENCE_RESPONSE } from '../app/translations';
+import { LinkItem } from '../common/links';
+import threatIntelligencePageImg from '../common/images/threat_intelligence.png';
```
Comment (PhilippeOberti) [inline x-pack/plugins/security_solution/public/threat_intelligence/links.ts:12]: I took a screenshot of the mock to create the image. I resized it to match the other images shown in this new navigation Explore page, but didn't round up the top left and right corners. Not sure what tool to use to do that 😆

File: x-pack/plugins/security_solution/common/constants.ts [hunk 4]
```diff
@@ -148,6 +149,7 @@ export const EVENT_FILTERS_PATH = `${MANAGEMENT_PATH}/event_filters` as const;
 export const HOST_ISOLATION_EXCEPTIONS_PATH =
   `${MANAGEMENT_PATH}/host_isolation_exceptions` as const;
 export const BLOCKLIST_PATH = `${MANAGEMENT_PATH}/blocklist` as const;
+export const THREAT_INTELLIGENCE_PATH = '/threat_intelligence' as const;
```
Comment (PhilippeOberti) [inline x-pack/plugins/security_solution/common/constants.ts:152]: not sure that's the url we want to use?

File: x-pack/plugins/security_solution/public/lazy_sub_plugins.tsx [hunk 5]
```diff
@@ -36,11 +35,11 @@ const subPluginClasses = {
   Users,
   Network,
   Kubernetes,
-
   Overview,
   Rules,
   Timelines,
   Management,
   LandingPages,
+  ThreatIntelligence,
```
Comment (PhilippeOberti) [inline x-pack/plugins/security_solution/public/lazy_sub_plugins.tsx:43]: this is what loads the external plugin in the Security Solution one

File: x-pack/plugins/security_solution/public/app/home/home_navigations.ts [hunk 6]
```diff
@@ -168,6 +169,13 @@ export const navTabs: SecurityNav = {
     disabled: false,
     urlKey: 'administration',
   },
+  [SecurityPageName.threatIntelligence]: {
+    id: SecurityPageName.threatIntelligence,
+    name: i18n.THREAT_INTELLIGENCE_RESPONSE,
+    href: APP_THREAT_INTELLIGENCE_PATH,
+    disabled: false,
```
Comment (maxcold) [inline x-pack/plugins/security_solution/public/app/home/home_navigations.ts:176]: means we can just disable it here if we need to before the release?

File: x-pack/plugins/security_solution/public/app/translations.ts [hunk 7]
```diff
@@ -131,3 +131,10 @@ export const NO_PERMISSIONS_MSG = (subPluginKey: string) =>
 export const NO_PERMISSIONS_TITLE = i18n.translate('xpack.securitySolution.noPermissionsTitle', {
   defaultMessage: 'Privileges required',
 });
+
+export const THREAT_INTELLIGENCE_RESPONSE = i18n.translate(
```
Comment (maxcold) [inline x-pack/plugins/security_solution/public/app/translations.ts:135]: I'm confused with the RESPONSE part here, should it be just `THREAT_INTELLIGENCE`? if you added it as DETECTIONS_RESPONSE has it, it's just the name of the feature, Detections & Response

File: x-pack/plugins/security_solution/public/threat_intelligence/links.ts (hunk 3 above)
Comment (maxcold) [reply x-pack/plugins/security_solution/public/threat_intelligence/links.ts:12]: I believe you can do it in Figma, should be quite straightforward. We can also ask Fernanda to prepare the image ofc

**Comment by maxcold** [comment]
LGTM, works as expected! what's the plan - wait for https://github.com/lgmys/kibana/pull/5 to be merged, rebase and merge this one? or do we want to wait till we are in upstream?

File: x-pack/plugins/threat_intelligence/public/plugin.tsx [hunk 8]
```diff
@@ -5,23 +5,18 @@
...
-        const [coreStart] = await core.getStartServices();
-        return renderApp(coreStart, params);
-      },
-    });
-  }
-  public start(_core: CoreStart) {
+  public setup(core: CoreSetup): ThreatIntelligencePluginSetup {
     return {};
   }
+  public start(_core: CoreStart): ThreatIntelligencePluginStart {
```
Comment (lgmys) [inline x-pack/plugins/threat_intelligence/public/plugin.tsx:18]: @PhilippeOberti  how does `start` differ from `setup`?

File: x-pack/plugins/security_solution/public/threat_intelligence/pages/threat_intelligence.tsx [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+const ThreatIntelligence = () => {
+  const services = useKibana().services;
+  const { threatIntelligence } = services;
+  const ThreatIntelligenceComponent = threatIntelligence.getComponent();
+
+  return (
+    <SecuritySolutionPageWrapper noPadding>
+      <ThreatIntelligenceComponent services={services} />
+      <SpyRoute pageName={SecurityPageName.threatIntelligence} />
```
Comment (lgmys) [inline x-pack/plugins/security_solution/public/threat_intelligence/pages/threat_intelligence.tsx:22]: :D:D

**Review by lgmys — APPROVED** [review]

File: x-pack/plugins/security_solution/public/app/translations.ts (hunk 7 above)
Comment (PhilippeOberti) [reply x-pack/plugins/security_solution/public/app/translations.ts:135]: yeah it definitely should just be `THREAT_INTELLIGENCE`, sorry about that!

File: x-pack/plugins/security_solution/public/app/home/home_navigations.ts (hunk 6 above)
Comment (PhilippeOberti) [reply x-pack/plugins/security_solution/public/app/home/home_navigations.ts:176]: that's something I'd like to look into today. I was hoping Elastic/Kibana would have some sort of feature flag functionality like [LauchDarkly](https://launchdarkly.com/?utm_source=google&utm_medium=cpc&utm_campaign=NA_-_Search_-_Brand_-_zzz&utm_term=what%20is%20launch%20darkly&utm_content=&obility_id=118354835243-604440797749&_bt=604440797749&_bm=e&_bn=g&gclid=Cj0KCQjwlK-WBhDjARIsAO2sErRBn-x_v_v_1h55klKnBLZXiIIto7gEajzjJ92GPRgqmBeaN-qwKHIaAn7sEALw_wcB) but we don't then I'll find the best way to hide/disable everything in this PR, and we'll flip the switch only if we know for sure that we're going to make 8.4. Does this sound like a good plan?

Comment (PhilippeOberti) [reply x-pack/plugins/security_solution/public/app/home/home_navigations.ts:176]: setting disabled to true does what we expect, it shows the link in a disabled state. If that's what we want then yes we can do that.
For the new navigation I found a `hideWhenExperimentalKey` property that hides the entry in the new `Explore` screen so we're good for that one :)

**Comment by PhilippeOberti** [comment]
> LGTM, works as expected! what's the plan - wait for #5 to be merged, rebase and merge this one? or do we want to wait till we are in upstream?

@maxcold @lgmys I think this is pretty pretty much ready, the only thing's left is to get the proper image for the Explore link (I posted [on Slack](https://elastic.slack.com/archives/C03FNQE31LY/p1657553008882289) for this)

I suggest we keep the link visible for now, to facilitate development and potential demos to other people in the team or outside?

**Comment by PhilippeOberti** [comment]
> 

New image with rounded corner uploaded, I'm merging this :)

**Pull request merged**